### PR TITLE
Bookmarks close

### PR DESCRIPTION
**Pull request by svidem**
Bookmarks now close when the video closes

### CODE REVIEW

File: ViAn/Filehandler/bookmark.cpp [hunk 1]
```diff
@@ -113,18 +112,6 @@ void Bookmark::create_file_path() {
...
-        // If file exists, try FRAMENR(X).tiff
-        path = QString(dir_path);
-        path.append("/");
-        path.append(QString::number(frame_number));
-        path.append("(");
-        path.append(QString::number(counter));
-        path.append(").tiff");
-        counter++;
-    }
-
```
Comment (parso619) [inline ViAn/Filehandler/bookmark.cpp]: This should not be removed.

File: ViAn/Filehandler/bookmark.h [hunk 2]
```diff
@@ -8,16 +8,17 @@
...
  * Bookmark class is used for storing bookmarks, i.e. user
  * marked points in a video and an associated frame.
  */
-class Bookmark : Saveable{
+class Bookmark : Saveable{    
 public:
     Bookmark(int frame_nbr, QImage frame, QString dir_path, QString string);
     Bookmark();
-    int get_frame_number();
+    int get_frame_number();
```
Comment (parso619) [inline ViAn/Filehandler/bookmark.h]: These extra spaces/tabs should be removed too.

File: ViAn/Filehandler/filehandler.cpp [hunk 3]
```diff
@@ -505,7 +505,7 @@ void FileHandler::add_dir(ID dir_id, QDir dir){
 /**
  * @brief FileHandler::proj_equals
  * @param proj
- * @param proj2
+    * @param proj2
```
Comment (parso619) [inline ViAn/Filehandler/filehandler.cpp]: Extra spaces, again.

File: ViAn/GUI/mainwindow.cpp [hunk 4]
```diff
@@ -781,8 +774,18 @@ void MainWindow::on_action_delete_triggered() {
...
                 this->file_handler->remove_video_from_project(my_project->id, my_item->id); // Remove video from project
             } else if (my_item->type == TYPE::PROJECT) {
+                for(int i = 0; i < my_item->childCount(); i++) {
+                    QTreeVideoItem* v = (QTreeVideoItem*)my_item->child(i);
+                    for(auto it = v->bookmarks.begin(); it != v->bookmarks.end(); it++){
+                        bookmark_view->remove_bookmark(v->id, *it);
+                    }
+                }
                 this->file_handler->delete_project(my_item->id);
             }
```
Comment (DanielPerssonProos) [inline ViAn/GUI/mainwindow.cpp]: Could be moved to separate function

File: ViAn/GUI/qtreeitems.h [hunk 5]
```diff
@@ -5,16 +5,17 @@
...
+#include "bookmarkview.h"
 #define TEXT_LENGTH 15
-
+typedef int ID;
 enum TYPE {PROJECT, VIDEO};
 
-
 class MyQTreeWidgetItem: public QTreeWidgetItem {
 public:
-    explicit MyQTreeWidgetItem(TYPE type, QString name = "", int id = 0);
```
Comment (DanielPerssonProos) [inline ViAn/GUI/qtreeitems.h]: id should not be set to a value per default that could be mistaken for a valid id.

File: ViAn/GUI/bookmarkview.cpp [hunk 6]
```diff
@@ -24,9 +24,23 @@ int BookmarkView::get_num_bookmarks() {
  * Adds a bookmark to the bookmark view.
  * @param bookmark Bookmark to add.
  */
-void BookmarkView::add_bookmark(Bookmark* bookmark) {
+int BookmarkView::add_bookmark(ID video_id, ID bookmark_id, Bookmark* bookmark) {
```
Comment (henpe071) [inline ViAn/GUI/bookmarkview.cpp]: Shouldn't it have a "return" if you say it returns an int?

**Pull request merged**